src/Middleware/AnonymousPageDecorationMiddleware.php

```diff
@@ -0,0 +1,68 @@
+<?php
+namespace LORIS\Middleware;
+use \Psr\Http\Message\ServerRequestInterface;
+use \Psr\Http\Message\ResponseInterface;
+use \Psr\Http\Server\MiddlewareInterface;
+use \Psr\Http\Server\RequestHandlerInterface;
+
+class AnonymousPageDecorationMiddleware implements MiddlewareInterface {
+    protected $JSFiles;
+    protected $CSSFiles;
+    protected $Config;
+    protected $BaseURL;
+
+    public function __construct(string $baseurl, \NDB_Config $config, array $JS, array $CSS) {
+        $this->JSFiles = $JS;
+        $this->CSSFiles = $CSS;
+        $this->Config = $config;
+        $this->BaseURL = $baseurl;
+    }
+
+    /**
+     * Displays a page to an anonymous (not logged in) user.
+     *
+     * @param array $get  The GET parameters from the request.
+     * @param array $post The POST parameters from the request.
+     *
+     * @return string the page content
+     */
+    public function process(ServerRequestInterface $request, RequestHandlerInterface $handler) : ResponseInterface {
+        // Basic page outline variables
+        $tpl_data = array(
+            'study_title' => $this->Config->getSetting('title'),
+            'baseurl'     => $this->BaseURL,
+            'currentyear' => date('Y'),
+            'sandbox'     => ($this->Config->getSetting("sandbox") === '1'),
+        );
+
+        // I don't think anyone uses this. It's not really supported
```

Comment:
It is used by surveys

Reply:
Surveys don't use this page decoration middleware, though.